### ISSUE TITLE
General refactoring

### ISSUE THREAD

**Issue by janluk00** [janluk00/school-management-app#19]
improve:
- `class accessibility` - Review and modify class access levels. Not all classes should be public.
- `encapsulation` - Limit the use of setters and getters.
- `code clean-up` - simplify complex methods, remove unused code and imports, improve overall readability.
